### ISSUE TITLE
Merchants - fade/block items for rubies

### ISSUE THREAD

**Issue by FrutyX** [DinoDevs/GladiatusCrazyAddon#355]
So with the server merges, there is a little fear that the auctions will be unusable and me and perhaps many others will end up buying food from merchants, I swear I thought this feature is there, but it's not (or did I miss it? Can't see it anywhere).

You can fade items that you can't afford, but I can afford them, so it's useless. So, simply, it could happen, that I buy some food for rubies by accident, so I suggest to add an option to fade items for rubies, or possibly even hide/block them, just whatever will prevent me from buying items for rubies.

**Comment by FilipSla** [comment]
This is a little bit agressive, but working quite nice.
`const itemsRub = document.querySelectorAll('.ui-droppable-grid .ui-draggable.ui-droppable');
for (let i = 0; i < itemsRub.length; i++) {
    if (containAtribute(itemsRub[i])) {
        itemsRub[i].style.backgroundColor = "red";
        itemsRub[i].setAttribute('data-gca-flag_double-click-event', false);
    }
}`

` function containAtribute(e) {
    if (e.getAttribute('data-tooltip').includes("icon_rubies")) {
        return e.getAttribute('data-tooltip');
    }
} ` 
![image](https://user-images.githubusercontent.com/44546766/175833999-8a45f132-25e7-4800-9e02-0444decae9dd.png)

**Comment by FrutyX** [comment]
I had a working prototype, I implemented it, even added it in the settings and languages, and it completely broke... ah yes, javascript.

I think @GreatApo is the right man for the job, just use the `fadeUnaffordableItems` from **merchants.js** and make it a new one, and to take something from Filip, do not use only opacity to fade it, I would also add an red background to these items, ruby-ish color :D :).

**Comment by FrutyX** [comment]
Correction, I could not live with my own failure, I think I now know where I did a mistake and will most likely create pull request, if things go well.

**Comment by FrutyX** [comment]
I know, I was taking it as a "feature", I guess we need more opinions, however your solution is good too, I like it.

**Comment by FilipSla** [comment]
Really nice, hadn´t try it yet but looks quite nice to me. As I sad I made it a bit agressive - blocked double click.

**Comment by GreatApo** [comment]
> Really nice, hadn´t try it yet but looks quite nice to me. As I sad I made it a bit agressive - blocked double click.

I look at the code now and you already can't buy items with rubies by double click...

**Comment by FrutyX** [comment]
Really? I said I may be blind but I did not see such a feature mentioned anywhere, anyways, I wouldn't said that what I/we did is bad, it's good for the view.

**Comment by FilipSla** [comment]
> > Really nice, hadn´t try it yet but looks quite nice to me. As I sad I made it a bit agressive - blocked double click.
> 
> I look at the code now and you already can't buy items with rubies by double click... You are suppose to just get a notification.

Yes, it was my solution for missclicks. But you can still buy them by dragging manually to the inventory.

**Comment by GreatApo** [comment]
I think this is a nice solution and we can also not interfere a lot with the game's monetization :)

**Comment by FrutyX** [comment]
Yea I suppose you are right, thanks for the improvements, it's good and useful.

**Comment by GreatApo** [comment]
> Yea I suppose you are right, thanks for the improvements, it's good and useful.

Thank you for scripting it. I changed the options translation, can you update yours please?

**Comment by SleepingShadow-bit** [comment]
Awesome guys. This saves finite rubies =)

**Comment by GreatApo** [comment]
Fruty is ready to hant down all translators :P

**Comment by FrutyX** [comment]
Implemented in v4.3.6, closing, re-open if needed.
